python/tank/pipelineconfig.py

```diff
@@ -761,30 +761,33 @@ def execute_post_install_bundle_hook(self, bundle_path):
     def _preprocess_descriptor(self, descriptor_dict):
         """
         Preprocess descriptor dictionary to resolve config-specific
-        constants and directives such as {PIPELINE_CONFIG}.
+        constants and directives such as {PIPELINE_CONFIG} and
+        {CONFIG_ROOT}
 
         :param descriptor_dict: Descriptor dict to operate on
         :returns: Descriptor dict with any directives resolved.
         """
 
-        if descriptor_dict.get("type") == "dev":
-            # several different path parameters are supported by the dev descriptor.
-            # scan through all path keys and look for pipeline config token
+        # Not a path or dev descriptor, early out.
+        if descriptor_dict.get("type") not in ["dev", "path"]:
```

Comment:
Here we are handling the new token. We're also fixing the code so PIPELINE_CONFIG and CONFIG_FOLDER can be used for both `path` and `dev` descriptors.